### PR TITLE
attempt to add Portage(emerge) and some fixes for apk

### PR DESCRIPTION
**Pull request by tanmatsu-cmd**
Hey again!

This is an attempt to add Portage (emerge),  [Gentoo](https://gentoo.org/)'s package manager. Alongside some minor fixes on apk.

## PORTAGE NOTES

- Listing installed packages is only doable via an external program (`qlist -IRv`) *or* using coreutils 

- Displaying a count of all packages is also only doable via an external program (`eix --world | tot`) *or* using coreutils

- Syncing, if desired, could be also done via `emerge --sync`

### CODE REVIEW

File: cpm [hunk 1]
```diff
@@ -144,6 +144,20 @@ _apt() {
     esac
 }
 
+_portage() {
+    case "$OP" in
+	install) $SUDO emerge -atv "$@";;
+	remove)  $SUDO emerge -avc "$@";;
+	list)    echo "qlist -IRv";;
+	count)   echo "eix --world | tot";;
```
Comment (willeccles) [inline cpm:152]: What's up with the `echo` here?

**Comment by willeccles** [comment]
> Listing installed packages is only doable via an external program (qlist -IRv) or using coreutils

I doubt this could only be done with coreutils or an external program. However, if this external program is likely to be installed, it may be worth using. Could you elaborate?

**Comment by tanmatsu-cmd** [comment]
> What's up with the `echo` here?

It's put as a placeholder. Portage does not have any programs for listing programs by itself.

> I doubt this could only be done with coreutils or an external program. However, if this external program is likely to be installed, it may be worth using. Could you elaborate?

Said programs needed would be `qlist` and `eix`, most Gentoo users are likely to install them but if needed I could make cpm possibly check for them too. As for why this or coreutils would be the only way is explained above.

**Comment by willeccles** [comment]
Okay, I think what should be done is to make those two cases multiple lines, so that you end up with something like this:

```bash
list)
    if has qlist; then
        # do that here
    elif has otherthing; then
        # do other things here
    else
        >&2 echo "Could not find program1 or program2"
        exit 1
    fi
    ;;
```

**Comment by tanmatsu-cmd** [comment]
> Okay, I think what should be done is to make those two cases multiple lines, so that you end up with something like this:
> 
> ```shell
> list)
>     if has qlist; then
>         # do that here
>     elif has otherthing; then
>         # do other things here
>     else
>         >&2 echo "Could not find program1 or program2"
>         exit 1
>     fi
>     ;;
> ```

Good idea! Will get started on it.

File: cpm [hunk 2]
```diff
@@ -144,6 +144,30 @@ _apt() {
...
+	list)    if has qlist; then
+		 	qlist -IRv
+	 	 else
+		 	>&2 echo "Could not find program qlist"
+		 fi
+		 ;;
+	count)   if has eix; then
+			eix --world | tot
+		 else
+			>&2 echo "Could not find program eix"
```
Comment (willeccles) [inline cpm:160]: Add `exit 1` to these `else` statements.

File: cpm [hunk 3]
```diff
@@ -144,6 +144,30 @@ _apt() {
...
 }
 
+_portage() {
+    case "$OP" in
+	install) $SUDO emerge -atv "$@";;
+	remove)  $SUDO emerge -avc "$@";;
+	list)    if has qlist; then
+		 	qlist -IRv
+	 	 else
+		 	>&2 echo "Could not find program qlist"
```
Comment (willeccles) [inline cpm:154]: Remove "program," just leave "Could not find qlist"

**Review by willeccles — CHANGES_REQUESTED** [review]

**Comment by willeccles** [comment]
In the future, use spaces instead of tabs. However, I will merge this and update it myself, as I am going to add a modeline to all my projects' source files from now on anyway.

**Pull request merged**